x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -85,1106 +85,1102 @@ describe('CredentialsLogic', () => {
-
-      CredentialsLogic.actions.fetchDetails();
-      expect(HttpLogic.values.http.get).toHaveBeenCalledWith('/api/app_search/credentials/details');
-      await promise;
-      expect(CredentialsLogic.actions.setCredentialsDetails).toHaveBeenCalledWith(
-        credentialsDetails
-      );
+      it('handles errors', async () => {
+        mount();
+        const promise = Promise.reject('An error occured');
+        (HttpLogic.values.http.get as jest.Mock).mockReturnValue(promise);
+
+        CredentialsLogic.actions.fetchDetails();
+        try {
+          await promise;
+        } catch {
+          expect(flashAPIErrors).toHaveBeenCalledWith('An error occured');
+        }
+      });
     });
 
-    it('handles errors', async () => {
-      mount();
-      const promise = Promise.reject('An error occured');
-      (HttpLogic.values.http.get as jest.Mock).mockReturnValue(promise);
+    describe('deleteApiKey', () => {
+      const tokenName = 'abc123';
 
-      CredentialsLogic.actions.fetchDetails();
-      try {
+      it('will call an API endpoint and set the results with the `onApiKeyDelete` action', async () => {
+        mount();
+        jest.spyOn(CredentialsLogic.actions, 'onApiKeyDelete').mockImplementationOnce(() => {});
+        const promise = Promise.resolve();
+        (HttpLogic.values.http.delete as jest.Mock).mockReturnValue(promise);
+
+        CredentialsLogic.actions.deleteApiKey(tokenName);
+        expect(HttpLogic.values.http.delete).toHaveBeenCalledWith(
+          `/api/app_search/credentials/${tokenName}`
+        );
         await promise;
-      } catch {
-        expect(flashAPIErrors).toHaveBeenCalledWith('An error occured');
-      }
+        expect(CredentialsLogic.actions.onApiKeyDelete).toHaveBeenCalledWith(tokenName);
+      });
+
+      it('handles errors', async () => {
+        mount();
+        const promise = Promise.reject('An error occured');
+        (HttpLogic.values.http.delete as jest.Mock).mockReturnValue(promise);
+
+        CredentialsLogic.actions.deleteApiKey(tokenName);
+        try {
+          await promise;
+        } catch {
+          expect(flashAPIErrors).toHaveBeenCalledWith('An error occured');
+        }
+      });
     });
   });
 
-  describe('deleteApiKey', () => {
-    const tokenName = 'abc123';
-
-    it('will call an API endpoint and set the results with the `onApiKeyDelete` action', async () => {
-      mount();
-      jest.spyOn(CredentialsLogic.actions, 'onApiKeyDelete').mockImplementationOnce(() => {});
-      const promise = Promise.resolve();
-      (HttpLogic.values.http.delete as jest.Mock).mockReturnValue(promise);
-
-      CredentialsLogic.actions.deleteApiKey(tokenName);
-      expect(HttpLogic.values.http.delete).toHaveBeenCalledWith(
-        `/api/app_search/credentials/${tokenName}`
-      );
-      await promise;
-      expect(CredentialsLogic.actions.onApiKeyDelete).toHaveBeenCalledWith(tokenName);
-    });
+  describe('selectors', () => {
```

Comment:
[Not a blocking comment at all] I'm kinda 50/50 on selectors getting their own test block - I wonder if we should try to treat them the same as values where they get tested as a byproduct of actions? 🤷

That being said, I can't think of a better way to do this - I think I'd have to see various examples of how we test selectors overall to see if a pattern emerges. I think this is great for now! 👍

Reply:
I tool was not about this.